### PR TITLE
Make namespace aliasing consistent everywhere; enforce with clj-kondo

### LINKED ISSUE #19930: Clean up namespace requires
Clean up namespace requires.

Ideally there is a single alias for namespaces and they follow a convention whenever possible. I just added a few functions to `metabase.query-processor.util` and I see that the usages of that are

- `ag "metabase.query-processor.util :as qp.util" | wc -l` 9
- `ag "metabase.query-processor.util :as qputil" | wc -l` 21

We should audit for this type and settle on one standard. At one of these call sites the pattern stands out glaringly:
```clojure
[metabase.query-processor.async :as qp.async]
[metabase.query-processor.card :as qp.card]
[metabase.query-processor.pivot :as qp.pivot]
[metabase.query-processor.util :as qputil]
```

This would also be a good project for a newly onboarded person as it is largely mechanical could serve as a good introduction to the layout of the project.

### PR DESCRIPTION
**Pull request by bshepherdson**
See the table of aliases in .clj-kondo/config.edn

This is not currently enforced by CI, which is based on lint-config.edn rather
than .clj-kondo/config.edn

Notable patterns:
- `[metabase.api.foo :as api.foo]`
- `[metabase.models.foo :as foo]`
- `[metabase.query-processor.foo :as qp.foo]`
- `[metabase.server.middleware.foo :as mw.foo]`
- `[metabase.util.foo :as u.foo]`
- `[clj-http.client :as http]` and `[metabase.http-client :as client]`

Fixes #19930.

### CODE REVIEW

File: bin/common/src/metabuild_common/core.clj [hunk 1]
```diff
@@ -5,7 +5,7 @@
             [metabuild-common.files :as files]
             [metabuild-common.input :as input]
             [metabuild-common.misc :as misc]
-            [metabuild-common.output :as output]
+            [metabuild-common.output :as out]
```
Comment (dpsutton) [inline bin/common/src/metabuild_common/core.clj:8]: i'm surprised to see this change. The alias `output` seemed a better consistent choice.

Comment (dpsutton) [reply bin/common/src/metabuild_common/core.clj:8]: this seems the opposite of the `sh` -> `shell` below.

File: enterprise/backend/test/metabase_enterprise/audit_app/pages/common_test.clj [hunk 2]
```diff
@@ -2,7 +2,7 @@
   (:require [clojure.test :refer :all]
             [honeysql.core :as hsql]
             [metabase-enterprise.audit-app.interface :as audit.i]
-            [metabase-enterprise.audit-app.pages.common :as pages.common]
+            [metabase-enterprise.audit-app.pages.common :as common]
```
Comment (dpsutton) [inline enterprise/backend/test/metabase_enterprise/audit_app/pages/common_test.clj:5]: a pretty niche ns getting a `common` alias seems off to me

File: enterprise/backend/test/metabase_enterprise/sandbox/pulse_test.clj [hunk 3]
```diff
@@ -5,8 +5,8 @@
             [medley.core :as m]
             [metabase.email.messages :as messages]
             [metabase.models :refer [Card Pulse PulseCard PulseChannel PulseChannelRecipient]]
-            [metabase.models.pulse :as models.pulse]
-            [metabase.pulse :as pulse]
+            [metabase.models.pulse :as pulse]
+            metabase.pulse
```
Comment (dpsutton) [inline enterprise/backend/test/metabase_enterprise/sandbox/pulse_test.clj:9]: This is an error here i believe. We have `pulse/send-pulse!` in this namespace from `metabase.pulse/send-pulse!`. But this switches it to `metabase.models.pulse/send-pulse!` which does not exist.

**Comment by bshepherdson** [comment]
**Big shiny notice to reviewers**
These names don't need to be unique! So for example `schema.core` and `clojure.spec.alpha` as both mapped to `s`. That's not ideal in a sense, but there were dozens of uses for each but only one namespace with both.

The list is also not comprehensive, it's only aiming to (a) make consistent patterns for eg. `metabase.api.foo` and `metabase.models.foo` and (b) unify widely shared namespaces.
I'm sure I brought dozens more namespaces into this list than were really needed.

File: bin/common/src/metabuild_common/core.clj (hunk 1 above)
Comment (dpsutton) [reply bin/common/src/metabuild_common/core.clj:8]: no bikeshedding. i'm not looking for all of these to match my expectations. cohesive whole is the goal, not my personal. I was just surprised.

File: enterprise/backend/test/metabase_enterprise/sandbox/pulse_test.clj (hunk 3 above)
Comment (dpsutton) [reply enterprise/backend/test/metabase_enterprise/sandbox/pulse_test.clj:9]: oh i see you used the fully qualified name below. I actually really like that.

**Review by dpsutton — APPROVED** [review]
Big fan. I had no idea it would be such a big changeset. This is awesome. Want to call out a particularly clever solution:

`metabase.models.pulse` and `metabase.pulse`. Now the model is `pulse` and the top level namespace is just used as its fqn name `metabase.pulse/send-pulse!`. Nice solution. Easy to forget that not everything needs an alias.

**Pull request merged**